### ISSUE TITLE
Allow to open streams with zero length buffer and observe these on the remote peer

### ISSUE THREAD

**Issue by normanmaurer** [cloudflare/quiche#731]
At the moment when `Connection::send_stream(...)` is called with a zero length data buffer (no fin flag) it will succeed but it will not be "observed" on the remote peer. I think this is because you optimize for the amount of data that is written on the wire. 

That said I think it sometimes would still be useful to also send this empty stream frame over the wire in the case when you want to open a stream from one peer and start writing from the other without the need to receive any data first. I know you could argue that for such a use-case you may be better off to just open the stream from the other peer in the first place but still I think this is a valid use-case which matches what should be possible based on the quic RFC.

**Comment by LPardue** [comment]
@normanmaurer  would you still want this?

> "send this empty stream frame over the wire in the case when you want to open a stream from one peer and start writing from the other without the need to receive any data first

**Comment by normanmaurer** [comment]
@LPardue yes it would be nice to allow to force it send an empty stream_frame... sorry for the late response
